week10/src/pages/Detail.jsx

```diff
@@ -0,0 +1,392 @@
+import { useEffect, useState } from "react";
+import { useParams, useNavigate } from "react-router-dom";
+import { GetPostDetailApi, EditPostApi, DeletePostApi } from "../api/post";
+import {
+    postHeart,
+    deleteHeart,
+    getHeartNum,
+    checkMyClick,
+} from "../api/heart";
+import { deleteComments, getComments, postComments } from "../api/comment";
+import { JWTtest } from "../api/user";
+import styled from "styled-components";
+
+import heart from "../img/heart.png";
+
+const Detail = () => {
+    const { postId } = useParams();
+    const [details, setDetails] = useState(null);
+    const [isEdit, setIsEdit] = useState(false);
+    const [post, setPost] = useState({ title: "", content: "" });
+    const [heartNum, setHeartNum] = useState(0);
+    const [myHeart, setMyheart] = useState(false);
+    const [myComments, setMyComments] = useState("");
+    const [commentsList, setCommentsList] = useState(null);
+    const [myAccount, setMyAccount] = useState("");
+    const nav = useNavigate();
+
+    useEffect(() => {
+        async function getDetails() {
+            const res = await GetPostDetailApi(postId);
+            setDetails(res);
+            const res2 = await JWTtest();
+
+            setMyAccount(res2.nickname);
+        }
+
+        async function getHeartNumInDetail() {
+            const num = await getHeartNum(postId);
+            setHeartNum(num);
+        }
+
+        async function getCommentsList() {
+            const res = await getComments(postId);
+            setCommentsList(res);
+        }
+
+        async function didIClickHeart() {
+            const HeartHistory = await checkMyClick(postId);
+            setMyheart(HeartHistory);
+        }
+
+        if (!localStorage.getItem("efubtoken")) {
+            alert("로그인 후 이용해주세요.");
+            nav("/register");
+        } else {
+            getDetails();
+            getHeartNumInDetail();
+            getCommentsList();
+            didIClickHeart();
+        }
+    }, [postId]);
```

Comment:
```suggestion
        if (!localStorage.getItem("efubtoken")) {
            alert("로그인 후 이용해주세요.");
            nav("/register");
        } else {
            getDetails();
            getHeartNumInDetail();
            getCommentsList();
            didIClickHeart();
        }
    }, []);
```
useEffect 내부에는 이렇게 함수 호출문만 남기고 정의문은 모두 바깥으로 빼주면 좀더 가독성 측면에서 나은 코드가 될 것 같습니다.
추가로 현재 페이지에서는 postId가 변경되는 일이 발생하지 않으니 (다른 페이지로 이동 위해서는 반드시 뒤로가기 해야함, 상세 페이지에서 다른 id의 상세 페이지로 이동하는 루트 없음) 굳이 의존성 배열에 postId를 넣어줄 필요는 없습니다. 첫 렌더 시 한번만 API 호출이 발생하면 되기 때문에 의존성 배열은 빈 배열로 설정해주세요.